src/CADL.Extension/Emitter.Csharp/src/emitter.ts

```diff
@@ -553,18 +572,28 @@ function loadOperation(
         program: Program,
         response: HttpOperationResponse,
         resourceOperation?: ResourceOperation
-    ): OperationResponse | undefined {
+    ): { operationResponse?: OperationResponse; paging?: OperationPaging } {
```

Comment:
I'm not sure I understand why `paging` is part of `loadOperationResponse`? It looks like code below `/* handle paging */` can be extracted into separate function

Reply:
All the paging information is stored in response, so I handle together in loadOperationResponse.

Reply:
No, paging information is stored in the [`InputOperation`](https://github.com/Azure/autorest.csharp/blob/feature/v3/src/CADL.Extension/Emitter.Csharp/src/type/InputOperation.ts#L32):

Reply:
If `operation.responses[0].responses[0]` contains paging information, then `operation.responses` can have only one element and `operation.responses[0].responses` also can have only one element. Any other configuration is not supported by autorest.csharp.

Reply:
Hello @AlexanderSher  I separated paging handling from loadOperationResponse.